uaclient/tests/test_config.py

```diff
@@ -18,6 +18,45 @@
 M_PATH = 'uaclient.entitlements.'
 
 
+class TestEntitlements:
+
+    def test_entitlements_property_keyed_by_entitlement_name(self, tmpdir):
+        """Return machine_token resourceEntitlements, keyed by name."""
+        cfg = UAConfig({'data_dir': tmpdir.strpath})
+        token = {
+            'machineTokenInfo': {'contractInfo': {'resourceEntitlements': [
+                {'type': 'entitlement1', 'entitled': True},
+                {'type': 'entitlement2', 'entitled': True}]}}}
+        cfg.write_cache('machine-token', token)
+        expected = {
+            'entitlement1': {
+                'entitlement': {'entitled': True, 'type': 'entitlement1'}},
+            'entitlement2': {
+                'entitlement': {'entitled': True, 'type': 'entitlement2'}}}
+        assert expected == cfg.entitlements
+
+    def test_entitlements_use_machine_access_when_present(self, tmpdir):
+        """Return specific machine-access info if present."""
```

Comment:
This is orthogonal to this PR, but for my edification: when would we expect machine-access and machine-token to be out of sync?

Reply:
machine-access *should* always be present for every enabled entitlement. It's a bug if we don't have a machine-access-<service>,json for every entitled service. The only drift could have been introduced if contract entitlements change from entitled to not-entitled. But, in that case we clean up machine-access files as deltas are processed.

Reply:
Thanks!